feat: implement Stripe billing integration with pricing table and free trial flow

valverdesolera opened the pull request:
## Summary

Implements the full Stripe billing integration for the platform, including a custom pricing table, free trial support, add-on purchases (channels, members, AI tokens), proration previews, and subscription lifecycle management. This also includes Whapi WhatsApp channel improvements and various billing UI refinements.

## Changes

### Billing & Subscription Core
- Stripe subscription creation, checkout, and billing portal integration
- Free trial flow with configurable trial days and automatic expiration handling
- Subscription lifecycle management (active, trialing, canceled, past due states)
- Monthly/yearly price selection with correct plan switching
- Cancel-at-period-end and scheduled cancellation date display
- Billing portal return URL configuration
- Idempotent Stripe customer creation

### Pricing Table
- Custom pricing table component with Starter and Custom plans
- Trial status indicators and cancellation date display
- Dynamic plan button logic based on subscription state
- Disabled add-on purchases during free trial and past-due states

### Add-ons & Proration
- Channel and member purchase flows with proration previews
- Credit preview on member removal
- Channel deletion proration
- Proration message suppressed when no extra agents/members exist

### AI Token Credits
- AI token credit authorization and enforcement
- Monthly token reset on subscription renewal
- Initial token limit initialization
- Insufficient tokens banner notification
- Improved token credit reset logic for Stripe billing

### Billing UI
- Current subscription and costs breakdown card
- Billing limits card with usage tracking
- Subscription detail items (plan, seat count, renewal date, status, end date)
- Removed Captain section, Chat With Us section from billing page
- Commented out Agency Force Assist training card

### Whapi WhatsApp Integration
- Webhook-driven Whapi reconnection flow
- Whapi disconnection handling
- Whapi enabled by default for new channels
- User logout on channel deletion
- Reconnection success state cleanup

### Bug Fixes
- Race condition fix on fast member deletion
- Race condition fix on quick channel deletion
- Billing dates unified across all sections
- Fixed onboarding redirect preventing access to billing and other settings pages
- Fixed billing page redirect for non-cloud deployments

### Translations
- Added missing EN/ES translations for billing settings (pricing table, limits, subscription status, plans, add-ons)
- Added missing `account_name_already_exists` backend translation

## Technical Notes
- Billing data stored in account `custom_attributes` (plan_name, subscription_status, stripe_customer_id, etc.)
- Subscription and limits fetched via Vuex store actions (`accounts/subscription`, `accounts/limits`)
- Proration handled via Stripe API with preview calculations
- AI token limits enforced via `plan_usage_and_limits` enterprise module
- Custom `RetryableStripeError` exception for transient Stripe failures

## Testing
- Verify free trial signup → billing page shows trial status and expiration
- Test plan upgrade/downgrade via pricing table
- Verify add-on purchase (channels, members) shows proration preview
- Test subscription cancellation flow and scheduled cancellation display
- Verify AI token credit reset on subscription renewal
- Test Whapi channel reconnection via webhook
- Verify billing page loads correctly for non-cloud deployments

## Related
- Ticket: CU-86acm62nm

valverdesolera committed:
Updates

valverdesolera committed:
updates

Updates

valverdesolera committed:
updates

Updates

valverdesolera committed:
Add on Live Training

Live Training Add on Logic + Other Stripe logic retrieval

valverdesolera committed:
Updates

Updates

valverdesolera committed:
Updates

Updates

valverdesolera committed:
Updates

Updates

valverdesolera committed:
updates

updates

valverdesolera committed:
updates

updates

valverdesolera committed:
removal of md documents

Removal of md documents

valverdesolera committed:
Updates

Updates

valverdesolera committed:
Billing Preview

Billing Preview indicating the credit line

valverdesolera committed:
Updates

Updates

valverdesolera committed:
Using Stripe Limits

Using Stripe Limits

valverdesolera committed:
Current Subscription & Costs

Invoice is getting the correct information and time period

valverdesolera committed:
Current Subscription & Costs Including things

Now it takes the information from the metadata and includes it in the Current Subscription & Costs

valverdesolera committed:
Inboxes Subsection

Inboxes subsection now shows the Extra inboxes

valverdesolera committed:
AI Token Credits Authorization

Improves the authorization because it was having 401 Auth issues

valverdesolera committed:
Credit Preview on Member Removal

Not when previewing the deletion of a member, it shows the correct credit amount

valverdesolera committed:
Create Channel Flow Adding Members Purchase

Now in the creating a channel flow, it checks if the included members is over the limit and if so then proceeds to charge for it

valverdesolera committed:
Buy Channel Proration

When buying a channel, now it shows the correct proration charge

valverdesolera committed:
Proration Preview on Deleting a Channel

Deleting a Channel has now the correct proration preview

valverdesolera committed:
Quick Channel Deletion Race Condition Solved

Race Condition issue solved when deleting a channel. The included inboxes usage and extra inboxes needs to be loaded first

valverdesolera committed:
Race Condition Error Fixed when Deleting Members Fast

Solves the issue when deleting a member fast

valverdesolera committed:
Pricing Table Added Custom Plan and Aesthetic Improvements

Custom plan was added, also aesthetic improvements to the pricing table

valverdesolera committed:
Have only Starter & Custom Plan

Removal of Professional and Business Plan and ONLY have Starter Plan and Custom Plan

valverdesolera committed:
Banner Notification Warning and Insufficient Tokens

Banner notification for warning threshold and insufficient tokens

valverdesolera committed:
Update on Token Credits

Token Credits Update

valverdesolera committed:
Subscription Issue for Monthly/Yearly price selection solved

Solved the issue of selecting the monthly/yearly price so that it takes the correct time of subscription period

valverdesolera committed:
Pricing Table Trial being indicated correctly

Now the pricing table indicates the proper button when the user is on the free trial period

valverdesolera committed:
Cancels on due date text on the pricing table

When the user cancels, the cancellation date is the date when the subscription ends, therefore as an UI indication this needs to be added in the UI so that the user understands that the subscription is going to be cancelled

valverdesolera committed:
Return URL for Billing Portal

Adding the return URL for the billing portal to be chatscommerce

valverdesolera committed:
Dates from billing section are now all the same one

Previously dates were differing by +-1 but now they indicate the same date which is the one indicated on Stripe

valverdesolera committed:
Disable to buy extra add ons on free trial

Now on free trial the user can not buy add ons

valverdesolera committed:
Proration message not showing when there are no extra agents/members

Previously the proration message was shown even though there were no extra agents/members bought, which was wrong.

valverdesolera committed:
Text indicating that the AI Tokens reset monthly

Added text that indicates that the AI Tokens resets monthly

valverdesolera committed:
Reset of AI Tokens when subscription is renewed each month

Each month, when the user pays for the monthly subscription charge, the AI Tokens are renewed

valverdesolera committed:
Fix the Reset of Tokens

When advancing the time, it was not reseting the tokens correctly, but now it does correctly resets the tokens.

valverdesolera committed:
Fix the issue of different places in definitions of free trial days expiration

previously the value of the trial period days was defined in the backend and frontend  but now that was centralized in one place

valverdesolera committed:
Removed idempotency key for customer (NEEDS TO BE REVERTED)

This needs to be reverted

valverdesolera committed:
Monthly price being selected on Free Trial mode

Previosly the yearly price was selected by default, but now the monthly plan is the one selected, also via the lookup_key

valverdesolera committed:
Fix the issues of the user still being able to buy extra add ons

The users are still able to buy extra add ons when they are on the free trial

valverdesolera committed:
Open billing button made not visible when in the same page

the open billing button takes the user to the settings > billing page, but this button is still visible when in the settings > billing page, which makes no sense. This is fixed

valverdesolera committed:
Fixed issue when user was on past due so that can not buy extra add ons

Users now will not be able to buy extra add-ons when the payment is past due

valverdesolera committed:
Trial Updates and remove unnecessary and duplicate code

Now there is NO free trial plan, the user is assigned a Starter plan in Trialing mode.
Removed code inefficiencies when allowing customers to be able to buy add-ons or not.dd

valverdesolera committed:
Cancelation date is now indicated in the pricing table

Cancellation date is now indicated in the pricing table

valverdesolera committed:
Billing summary correctly shows 0 when subscription is due to cancellation

Now, when the user cancels the subscription, it shows the total of 0.00 and a cancellation date including when it is going to be cancelled

valverdesolera committed:
Billing when trialing is now correct

Now it shows the correct payment the user needs to make after trialing

valverdesolera committed:
When in Trial, user can update payment details

valverdesolera committed:
When user is on community plan. Button is shown Upgrade

Now instead of indicating to Start Trial on community plan, it indicates "Upgrade"

valverdesolera committed:
Initial token limit initialization

Now it correctly initializes the initial token limit for the store

valverdesolera committed:
Store ID correction

Now the store ID correctly matches the frontend with the AI Backend

valverdesolera committed:
Removing unnecessary .md file

valverdesolera committed:
Improve AI token credit reset logic for Stripe billing

Token credits are now only reset on subscription renewals, not on initial subscription creation or trial invoices. Added robust extraction of subscription ID from invoices to support various Stripe API versions and invoice types, and improved error handling during subscription retrieval.

valverdesolera committed:
Simplify PricingCard plan button logic

Refactored PricingCard.vue to only show 'Current Plan' for the user's active plan and 'Get in Touch' for custom plans. Removed complex plan comparison and subscription management logic from the button, delegating all other actions to the billing portal. Updated i18n files to add 'Current Plan' translation in English and Spanish.

valverdesolera committed:
Whapi Disconnection (Still in Progress)

Handling Whapi disconnection (still work in progress)

valverdesolera committed:
Enabling Whapi by default

Whapi needs to be enabled by default

apereiracvo commented on app/builders/account_builder.rb:
```diff
@@ -92,51 +91,35 @@ def create_user
-
-    # Validate required plan limits before proceeding
-    self.class.validate_plan_limits_for_free_trial(plan_limits, trial_plan_name)
+  def set_initial_plan_attributes
```
Recomiendo que estudiemos si estos datos lo podemos pasar en vez de `custom_attributes` a una tabla propiamente con los `custom_migrations`.

Ademas, a traves de `concerns` podemos extender el model `account` para que utilice los datos de esa tabla

apereiracvo commented on app/controllers/api/v2/accounts/subscriptions_controller.rb:
```diff
@@ -168,20 +128,30 @@ def portal_params
       account_id: current_account.id,
-      plan_name: custom_attrs['plan_name'] || 'free_trial',
-      subscription_status: custom_attrs['subscription_status'] || 'free_trial',
+      plan_name: custom_attrs['plan_name'] || 'starter',
```
seria bueno que estos valores como `starter` y demas sean constants tambien

apereiracvo commented on app/controllers/api/v2/accounts/billing/add_ons_controller.rb:
```diff
@@ -0,0 +1,382 @@
+  before_action :check_authorization
+  before_action :validate_add_on_type, only: [:update, :preview, :preview_purchase]
+
+  # GET /api/v2/accounts/:account_id/billing/add_ons
```
Seria. buenisimo ver si estamos (o si podemos) esquematizar los payloads y returns de estos APIs. definirlos cmo objetos, asi como ahcemos en python o typescript

apereiracvo commented on docs/ignore/BillingDataStorageArchitecture.md:
La carpeta correcta ignorada es `/docs/ignored/` no `/docs/ignore`. Eliminar estos

apereiracvo commented on docs/ignore/ClassicToFlexible.md:
La carpeta correcta ignorada es /docs/ignored/ no /docs/ignore. Eliminar estos

apereiracvo reviewed (changes requested):
Muy buen progress, pero hay que sincronizarlo con el fork-sync que podria tener sus cosas, entonces hacer eso con mucho cuidado.

Recordar:
- Los locales ahora deben agregarse en  ambos `locale` y `locale_custom`

valverdesolera committed:
Whapi reconnection state 2

valverdesolera committed:
Implement webhook-driven Whapi reconnection flow. Whapi working correctly

Adds a new API endpoint and frontend logic for initiating WhatsApp Whapi channel reconnection via webhook, with QR code delivery through ActionCable. Refactors Vue components and Vuex store to support websocket-based QR code handling, improves health status tracking, and ensures robust connection lifecycle management. Also updates i18n files for new status messages and enhances backend jobs and services for event broadcasting and QR code retrieval.

valverdesolera committed:
Green success text was extra and not needed when success of Whapi reconnection

valverdesolera committed:
Whapi improvements

Eliminates the legacy QR code polling endpoint and related frontend code, shifting to a websocket-based approach for QR delivery. Refactors error handling in the Whapi channels controller to use a helper method, and consolidates connection logic in the Whapi connection status service. Cleans up excessive logging and removes unused Vuex actions and API methods.

valverdesolera committed:
Loging out the user if channel is deleted

When a channel is deleted, first it logs out the user so that the session is not orphan in the WhatsApp of the user's phone

valverdesolera committed:
Live Training Forms

Live Trainings on Click Up

valverdesolera committed:
Idempotency of customer creation in Stripe Uncommented

valverdesolera committed:
Chat with us opens WhatsApp link

andvalsol committed:
Merge remote-tracking branch 'origin/development' into CU-86acm62nm_Chatwoot-Payment-Stripe-Review-Free-trial

# Conflicts:
#	app/builders/account_builder.rb
#	app/javascript/dashboard/api/ai_backend/entities/StoreEntity.js
#	app/javascript/dashboard/i18n/locale/en/inboxMgmt.json
#	app/javascript/dashboard/i18n/locale/en/settings.json
#	app/javascript/dashboard/i18n/locale/es/inboxMgmt.json
#	app/javascript/dashboard/i18n/locale/es/settings.json
#	app/javascript/dashboard/routes/dashboard/settings/agentBots/components/AgentBotModal.vue
#	app/javascript/dashboard/routes/dashboard/settings/billing/Index.vue
#	app/javascript/dashboard/routes/dashboard/settings/inbox/Settings.vue
#	app/javascript/dashboard/routes/dashboard/settings/inbox/channels/360DialogWhatsapp.vue
#	app/javascript/dashboard/routes/dashboard/settings/inbox/channels/Api.vue
#	app/javascript/dashboard/routes/dashboard/settings/inbox/channels/BandwidthSms.vue
#	app/javascript/dashboard/routes/dashboard/settings/inbox/channels/CloudWhatsapp.vue
#	app/javascript/dashboard/routes/dashboard/settings/inbox/channels/Line.vue
#	app/javascript/dashboard/routes/dashboard/settings/inbox/channels/Website.vue
#	app/javascript/dashboard/routes/dashboard/settings/inbox/channels/Whapi.vue
#	app/javascript/dashboard/routes/dashboard/settings/inbox/channels/emailChannels/ForwardToOption.vue
#	app/javascript/dashboard/store/modules/inboxes.js
#	app/services/forward_notification_service.rb
#	app/services/whatsapp/providers/whapi_service.rb
#	app/views/api/v1/models/_account.json.jbuilder
#	config/routes.rb
#	spec/builders/account_builder_spec.rb

andvalsol committed:
Redirect the user to the billing page. Add translations for english and spanish for the biling texts.

andvalsol committed:
Add missing translation in english: account_name_already_exists

andvalsol committed:
Add translations for RENEWS_ON, STATUS and ENDS_ON. Remove Captain and Chat with Us sections
